.github/workflows/check.yml

```
@@ -84,108 +97,28 @@ jobs:
-          token: ${{ secrets.CODECOV_TOKEN }}
-          verbose: true
-
-      - name: Send test report
-        if: ${{ always() && github.event_name == 'schedule' }}
-        uses: ./.github/actions/test-report
-        with:
-          name: '${{ github.job }}-${{ matrix.node-version }}'
-          result: ${{ job.status }}
+    # Fork PRs: gate on maintainer approval AND isolate secrets — only TRUNK_TOKEN
+    # reaches the test runner via the reusable workflow; DEPOT_TOKEN etc. are not passed.
+    environment: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.full_name != github.repository && 'external-pr' || '' }}
```

actionlint (1.7.12)

[error] when a reusable workflow is called with "uses", "environment" is not available. only following keys are allowed: "name", "uses", "with", "secrets", "needs", "if", and "permissions" in job "test"

(syntax-check)